src/transformers/generation_utils.py

```diff
@@ -2047,10 +2055,6 @@ def contrastive_search(
                 else:
                     this_peer_finished = True
 
-            # prepare model inputs
-            model_kwargs["past_key_values"] = past_key_values
```

Comment:
on the `_update_model_kwargs_for_generation` call above, `model_kwargs["past"]` is set with `past_key_values`. This is the variable that is used throughout the loop, not `model_kwargs["past_key_values"]` -- this line has no effect.